components/tab-panel/index.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * External dependencies
+ */
+import { partial, noop, omit } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+// import './style.scss';
+import { Component } from '@wordpress/element';
+import { default as withInstanceId } from '../higher-order/with-instance-id';
+import { NavigableMenu } from '../navigable-container';
+
+class TabButton extends Component {
+	constructor( { tabName, clickTab } ) {
+		super( ...arguments );
+		this.onClick = partial( clickTab, tabName );
+	}
```

Comment:
Per Riad's comment, let's get rid of this and turn TabButton into a function.

Reply:
K. So I'd be doing the partial application from TabPanel again? Would there be a way to avoid having an anonymous function or is it not important (still new to React, so have a lot of newbie questions).

Reply:
Correct, you’d create the function in TabPanel, either via partial() or with an anonymous function. In this case, there’s no avoiding creating some sort of function—a trade-off that can be seen in the block library too.